pantavisor.c

```diff
@@ -813,44 +813,47 @@ static void pv_remove(struct pantavisor *pv)
 
 static pv_state_t pv_shutdown(struct pantavisor *pv, shutdown_type_t t)
 {
-	char path[PATH_MAX];
-
 	if (!pv)
 		return PV_STATE_EXIT;
 
-	pv_log(INFO, "prepare %s...", shutdown_type_string(t));
-	wait_shell();
-
-	if (pv->state && pv_state_stop(pv->state))
-		pv_log(WARN, "stop error: ignoring due to %s",
+	pv_log(INFO, "preparing '%s'...", shutdown_type_string(t));
+	if (pv_config_get_system_init_mode() == IM_APPENGINE)
+		pv_log(INFO,
+		       "will not actually perform '%s' as we are in appengine mode",
 		       shutdown_type_string(t));
 
+	wait_shell();
+
 	if (REBOOT == t)
 		pv_wdt_start(pv);
```

Comment:
this shouldnt happen in appengine either imo.

Reply:
it will execute but won't start as we have disabled wdt from pantavisor.config in appengine. I'll change it here too though so we can prevent problems